src/lightly_train/_commands/train_task.py

```diff
@@ -260,15 +260,10 @@ def _train_task_from_config(config: TrainTaskConfig) -> None:
             step == 0
             or (step + 1) % no_auto(config.logger_args.log_every_num_steps) == 0
         )
-        is_val_step = (
-            step > 0
-            and (step + 1) % no_auto(config.logger_args.val_every_num_steps) == 0
-        )
-        is_save_ckpt_step = (
-            step > 0
-            and (step + 1) % no_auto(config.save_checkpoint_args.save_every_num_steps)
-            == 0
-        )
+        is_val_step = (step + 1) % no_auto(config.logger_args.val_every_num_steps) == 0
+        is_save_ckpt_step = (step + 1) % no_auto(
+            config.save_checkpoint_args.save_every_num_steps
+        ) == 0
```

Comment:
Removed the `step > 0` condition, otherwise validation / checkpoint saving cannot run after the first step.